src/engine/TransitivePathBase.cpp

```diff
@@ -40,6 +44,34 @@ TransitivePathBase::TransitivePathBase(
     variableColumns_[std::get<Variable>(rhs_.value_)] =
         makeAlwaysDefinedColumn(1);
   }
+  if (minDist_ == 0 && !lhs_.isBoundVariable() && !rhs_.isBoundVariable() &&
+      lhs_.isVariable() && rhs_.isVariable()) {
+    emptyPathBound_ = true;
+    auto makeInternalVariable = [](std::string_view string) {
+      return Variable{
+          absl::StrCat("?internal_property_path_variable_", string)};
+    };
+    // Dummy variables to get a full scan of the index.
+    auto x = makeInternalVariable("x");
+    auto y = makeInternalVariable("y");
+    auto z = makeInternalVariable("z");
+    auto allValues = ad_utility::makeExecutionTree<Union>(
+        qec,
+        ad_utility::makeExecutionTree<IndexScan>(
+            qec, Permutation::Enum::SPO,
+            SparqlTriple{TripleComponent{x}, PropertyPath::fromVariable(y),
+                         TripleComponent{z}},
+            activeGraphs),
+        ad_utility::makeExecutionTree<IndexScan>(
+            qec, Permutation::Enum::OPS,
+            SparqlTriple{TripleComponent{z}, PropertyPath::fromVariable(y),
+                         TripleComponent{x}},
+            activeGraphs));
+    auto uniqueValues = ad_utility::makeExecutionTree<Distinct>(
+        qec, QueryExecutionTree::createSortedTree(std::move(allValues), {0}),
```

Comment:
Doesn't the constructor of distinct perform the `createSortedTree` automatically?

Reply:
If not, it definitely should...
Fix this either here or (better) in a separate PR.